fpdf/fpdf.py

```diff
@@ -1266,17 +1267,116 @@ def rect(self, x, y, w, h, style=None):
             y (float): Ordinate of upper-left bounding box.
             w (float): Width.
             h (float): Height.
+            round_corners (tuple of str, tuple of fpdf.enums.Corner, bool): Optional draw a rectangle with round corners.
+            Possible values are:
+
+            *`TOP_LEFT`: a rectangle with round top left corner
+            *`TOP_RIGHT`: a rectangle with round top right corner
+            *`BOTTOM_LEFT`: a rectangle with round bottom left corner
+            *`BOTTOM_RIGHT`: a rectangle with round bottom right corner
+            *`True`: a rectangle with all round corners
+            *`False`: a rectangle with no round corners
             style (fpdf.enums.RenderStyle, str): Optional style of rendering. Possible values are:
 
             * `D` or empty string: draw border. This is the default value.
             * `F`: fill
             * `DF` or `FD`: draw and fill
         """
-        style = RenderStyle.coerce(style)
-        self._out(
-            f"{x * self.k:.2f} {(self.h - y) * self.k:.2f} {w * self.k:.2f} "
-            f"{-h * self.k:.2f} re {style.operator}"
-        )
+
+        if round_corners is not False:
+            self._draw_rounded_rect(x, y, w, h, style, round_corners)
+        else:
+            style = RenderStyle.coerce(style)
+            self._out(
+                f"{x * self.k:.2f} {(self.h - y) * self.k:.2f} {w * self.k:.2f} "
+                f"{-h * self.k:.2f} re {style.operator}"
+            )
+
+    def _draw_rounded_rect(self, x, y, w, h, style, round_corners):
+        min = h
+        r = (w - h) / 2
+
+        if r < 0:
+            r *= -1
+            min = w
+
+        if r == 0:
+            r = w / 5
+
+        if r >= min / 2:
+            r /= min
+
+        point_1 = point_8 = (x, y)
+        point_2 = point_3 = (x + w, y)
+        point_4 = point_5 = (x + w, y + h)
+        point_6 = point_7 = (x, y + h)
+        coor_x = [x, x + w, x, x + w]
+        coor_y = [y, y, y + h, y + h]
+
+        if round_corners is True:
```

Comment:
The logic in this `if` is good.
Now if `round_corners` is neither `True` nor `False`, a check should be made to detect invalid values.
Something like `round_corners = tuple(Corner.coerce(rc) for rc in round_corners)`

Reply:
I fixed that but i had a problem, when I call the method with just one corner the `for` split the `string` . To fix that i made the next code 
```
if len(round_corners) > 4: #here is the hardcode to detect when round_corners have just one corner
    round_corners = (round_corners,)
round_corners = tuple(Corner.coerce(rc) for rc in round_corners)
```
but I think is a bit hardcoded, do you think some other solution to solve this?

Reply:
> do you think some other solution to solve this?

Yes, `round_corners` should simply always be an iterable (a list or a tuple).
Single values should be passed this way: `round_corners=("BOTTOM_RIGHT",)`

Reply:
> I fixed that but i had a problem, when I call the method with just one corner the `for` split the `string` .

The pythonic fix would be to check for the type of the argument:

```python
    if isinstance(round_corners, (str, Corner)):  # single item
        round_corners = (Corner.coerce(round_corners),)
    else:
        round_corners = [Corner.coerce(rc) for rc in round_corners]
```

Btw.: Converting the last case into a tuple is just an unnecessary extra processing step with no real benefits.